src/EFCore.PG/Query/Internal/NpgsqlSqlNullabilityProcessor.cs

```diff
@@ -276,6 +278,16 @@ PostgresUnknownBinaryExpression postgresUnknownBinaryExpression
         return jsonTraversalExpression.Update(expression, newPath?.ToArray() ?? jsonTraversalExpression.Path);
     }
 
+    protected virtual SqlExpression VisitRowValueExpression(
+        PostgresRowValueExpression rowValueExpression,
+        bool allowOptimizedExpansion,
+        out bool nullable)
+    {
+        nullable = false;
+
+        return rowValueExpression;
```

Comment:
You don't want to visit inner SqlExpressions?

Reply:
Good catch, was an oversight.